src/library/scala/collection/immutable/Vector.scala

```diff
@@ -1187,13 +1182,9 @@ private[immutable] trait VectorPointer[T] {
         if (depth == 5) {
           display5 = new Array(32)
           display5((oldIndex >>  25) & 31) = display4
-          display4 = new Array(32)
-          display3 = new Array(32)
-          display2 = new Array(32)
-          display1 = new Array(32)
           depth +=1
         }
-        display4 = display5((newIndex >> 20) & 31).asInstanceOf[Array[AnyRef]]
```

Comment:
I would guess that this change is correct, but @TiarkRompf could confirm - I believe he is the original author.